logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java

```diff
@@ -50,6 +49,15 @@
   private final String mBaseLogsDir;
   private final Socket mSocket;
 
+  private static void setAcceptList(ValidatingObjectInputStream validatingObjectInputStream) {
+    validatingObjectInputStream.accept(java.util.Hashtable.class);
+    validatingObjectInputStream.accept("java.lang.*", "[Ljava.lang.*");
+    validatingObjectInputStream.accept(org.apache.log4j.spi.LoggingEvent.class);
```

Comment:
how would we know if there is a new class in a newer version of log4j that we would need to add to the accept list? is it possible to write a unit test?

if we cannot write a unit test to ensure forwards compatibility, are there better ways than to list out the specific classes? for example, is it safe to include everything in the org.apache.log4j.spi package? https://logging.apache.org/log4j/2.x/log4j-1.2-api/apidocs/org/apache/log4j/spi/package-tree.html

Reply:
This is something like the white list, and white list is requiring to add that per change.
Yes, an ut can be written to protect that.
Create an issue for this: https://github.com/Alluxio/alluxio/issues/14767